include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+
+/**
+ * Exactly like rp_runner_create, except that it is run without loading and
+ * saving the state of the pipeline before and after the execution, and that no
+ * flags are provided.
+ */
+rp_runner *rp_runner_create_memory_only(const char *pipeline_path,
+                                        uint64_t flags_count,
+                                        const char *flags[]);
+
+/**
+ *  Triggers the serialization of the pipeline on disk.
+ *  Returns 0 if a error happened, 1 otherwise.
+ *
+ *  runner cannot be null
+ */
+bool rp_runner_store_containers(rp_runner *runner);
+
+/**
+ * deletes the runner object and destroys all the resourced acquired by it.
+ */
+void rp_runner_destroy(rp_runner *runner);
+
+/**
+ * returns the amount of steps present in the runner.
+ *
+ * runner cannot be null
+ */
+uint64_t rp_runner_steps_count(rp_runner *runner);
+
+const inline uint64_t NO_KNOWN_STEP = UINT64_MAX;
+
+/**
+ * returns the index of the step with the provided name.
+ * returns NO_KNOWN_STEP if no step with such name existed.
+ *
+ * runner cannot be null
+ */
+uint64_t rp_runner_step_name_to_index(rp_runner *runner, const char *name);
+
+/**
+ * returns the step with the provided index, or null if not such step existed.
+ *
+ * runner cannot be null
+ */
+rp_step *rp_runner_get_step(rp_runner *runner, uint64_t index);
+rp_step *rp_runner_get_step_from_name(rp_runner *runner, const char *name);
+
+/**
+ * returns the step name
+ *
+ * step cannot be null
+ */
+const char *rp_step_get_name(rp_step *step);
+
+/**
+ *
+ * returns the container checkpoint associated to the provided container,
+ *
+ * step cannot be null
+ *
+ */
+rp_container_checkpoint *
+rp_step_get_checkpoint(rp_step *step, rp_container *container);
+
+rp_container_checkpoint *
+rp_step_get_checkpoint_from_name(rp_step *step, const char *name);
+
+/**
+ * returns the name of a container
+ */
+const char *
+rp_container_checkpoint_get_name(rp_container_checkpoint *container);
+
+/**
+ * return the kind with the provided name, null if no kind
+ * had the provided name.
+ *
+ * runner cannot be null
+ * kind_name cannot be null
+ */
+rp_kind *rp_kind_from_name(rp_runner *runner, const char *kind_name);
+
+/**
+ *
+ * return the name of a kind
+ *
+ * kind cannot be null
+ */
+const char *rp_kind_get_name(rp_kind *kind);
+
+/**
+ * creates a target from the provided info
+ *
+ * kind cannot be null
+ * path_components_count must be equal to the granularity depth of the kind
+ * path_components size must be equal to path_components_count
+ *
+ * returns 0 if a error was encountered, 1 otherwise.
+ */
+rp_target *rp_create_target(rp_kind *kind,
+                            int is_exact,
+                            uint64_t path_components_count,
+                            const char *path_components[]);
+
+/**
+ * Deletes the provided target, (THIS TARGET MUST BE ONE OBTAINED FROM
+ * rp_create_target)
+ */
```

Comment:
```suggestion
/**
 * Deletes the provided target
 *
 * \note WARNING: run this only on targets created through rp_create_target.
 */
```

Do we have a way to `assert` this?

Reply:
not really, it's a random heap object.